contrib/check-semver-feature.sh

```diff
@@ -0,0 +1,136 @@
+    generate_json_files_no_default_features "bitcoin_hashes"
+
+    # 4. bitcoin-units
+    generate_json_files_all_features "bitcoin-units"
+    generate_json_files_no_default_features "bitcoin-units"
+
+    # 5. bitcoin-io
+    generate_json_files_all_features "bitcoin-io"
+    generate_json_files_no_default_features "bitcoin-io"
+
+    # Check for API semver non-addivite cargo features on all the generated JSON files above.
+    run_cargo_semver_check "bitcoin"
+    run_cargo_semver_check "base58ck"
+    run_cargo_semver_check "bitcoin_hashes"
+    run_cargo_semver_check "bitcoin-units"
+    run_cargo_semver_check "bitcoin-io"
+
+    # Invoke cargo semver-checks to check for non-additive cargo features
+    # in all generated files.
+    check_for_non_addivite_cargo_features
+}
+
+# Run cargo doc with the cargo semver-checks rustdoc flags.
+# We don't care about dependencies.
+run_cargo_doc() {
+    RUSTDOCFLAGS="$RUSTDOCFLAGS" RUSTC_BOOTSTRAP=1 cargo doc --no-deps "$@"
+}
+
+# Run cargo semver-check
+run_cargo_semver_check() {
+    local crate="$1"
+
+    echo "Running cargo semver-checks for $crate"
+    # Hack to not fail on errors.
+    # This is necessary since cargo semver-checks will fail if the
+    # semver check fails.
+    # We check that manually later.
+    set +e
+    cargo semver-checks -v --baseline-rustdoc "$crate-no-default-features.json" --current-rustdoc "$crate-all-features.json" > "$crate--semver.txt" 2>&1
+    set -e
+}
+
+# The following function uses cargo doc to generate JSON files that
+# cargo semver-checks can use.
+# - no-default-features: generate JSON doc files with no default features.
+generate_json_files_no_default_features() {
+    local crate="$1"
+
+    echo "Running cargo doc no-default-features for $crate"
+    run_cargo_doc --no-default-features -p "$crate"
+
+    # replace _ for - in crate name.
+    # This is necessary since some crates have - in their name
+    # which will be converted to _ in the output file by cargo doc.
+    mv "target/doc/${crate//-/_}.json" "$crate-no-default-features.json"
+}
+# - all-features: generate JSON doc files with all features.
+generate_json_files_all_features() {
+    local crate="$1"
+
+    echo "Running cargo doc all-features for $crate"
+    run_cargo_doc --all-features -p "$crate"
+
+    # replace _ for - in crate name.
+    # This is necessary since some crates have - in their name
+    # which will be converted to _ in the output file by cargo doc.
+    mv -v "target/doc/${crate//-/_}.json" "$crate-all-features.json"
+}
+
+# Check if there are non-additive cargo features.
+# We loop through all the generated files and check if there is a FAIL
+# in the cargo semver-checks output.
+# If we detect a fail, we create an empty file non-additive-cargo.
+# If the following CI step finds this file, it will add:
+# 1. a comment on the PR.
+# 2. a label to the PR.
+check_for_non_addivite_cargo_features() {
+    for file in *semver.txt; do
```

Comment:
Corresponding change to the suggestion in my other comment, if you decide to follow it:
```suggestion
    for file in *additive-features.txt; do
```